pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -114,10 +126,57 @@ func (c *LatencyProfileController) sync(ctx context.Context, syncCtx factory.Syn
 		}
 	}
 
+	// In case a checkProfileRejection func was used, status will be updated accordingly
+	// ("" profile is also default profile, so "" -> anotherProfile: rejection can be handled)
+	if c.checkProfileRejectionFn != nil {
+		isRejected, rejectMsg, err := c.checkProfileRejectionFn(configNodeObj.Spec.WorkerLatencyProfile, uniqueRevisions)
+		if err != nil {
+			return err
+		}
+		// if profile transition is rejected, set status to progress=Unknown, complete=Unknown
+		// and degrade operator with suitable error message
+		if isRejected {
+			_, err = c.updateStatus(
+				ctx,
+				false, false, // not progressing, not complete
+				reasonLatencyProfileUpdateProhibited,
+				rejectMsg,
+			)
+			if err != nil {
+				return err
+			}
+			return fmt.Errorf(rejectMsg)
+		}
+	}
+
+	// in case profile was not rejected and,
+	// in case of empty workerlatency profile, set status Complete=False and pre-empt sync
+	if configNodeObj.Spec.WorkerLatencyProfile == "" {
+		_, err = c.updateStatus(
+			ctx,
+			false, true, // not progressing, complete=True
+			reasonLatencyProfileEmpty,
+			"latency profile not set on cluster",
+		)
+		return err
+	}
+
 	// For each revision, check that the configmap for that revision have correct arg val pairs or not
 	revisionsHaveSynced, err := c.matchRevisionsFn(configNodeObj.Spec.WorkerLatencyProfile, uniqueRevisions)
 	if err != nil {
-		return err
+		// during day-0: pre-empt
+		var revisionZeroErr *RevisionZeroError
```

Comment:
Is this important? If yes wouldn't it be easier to just return `revisionsHaveSyncedReason` or `revisionsHaveSyncedMessage` and resolve which msg to put into `reasonLatencyProfileUpdateTriggered` reason message . We can put that we are in cluster bootstrap phase into the message and just use the Update reason.

Also, tt seems a bit weird to me to use an error to signal this.

Reply:
Hmm, so I removed the `RevisionZeroError` and returning an extra variable with the `matchRevisionsFn` that contains the message about the revisions being synced or not, which is passed to the status update message; revision-0 has it's own message.

Reply:
+1